### PR TITLE
Switch to using requests.

### PR DESCRIPTION
**Pull request by kimishpatel**
Summary:
We are running into ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate
verify failed (_ssl.c:852), while using urrlib. Switching to requests instead.

Differential Revision: D17327704

### CODE REVIEW

**Comment by soumith** [comment]
> We are running into ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate
verify failed (_ssl.c:852), while using urrlib. Switching to requests instead.

Where are you running into this error. Requests seems to fix it, but is there an underlying reason this error started popping up suddenly?

**Comment by kimishpatel** [comment]
@soumith, these are broken in CI, 
pytorch_linux_xenial_cuda9_cudnn7_py3_NO_AVX2_test pytorch_linux_xenial_cuda9_cudnn7_py3_NO_AVX_NO_AVX2_test pytorch_linux_xenial_cuda9_cudnn7_py3_nogpu_test
pytorch_linux_xenial_cuda9_cudnn7_py3_test

@mingbowan did some digging down.
"so the problem is caused by openssl was compile into python interpreter differently, the default cert path was different (or pointing to wrong places) on the broken one"

**Comment by kimishpatel** [comment]
@soumith, I will wait to see if there is a resolution to the root cause.

**Comment by soumith** [comment]
For conda, @pietern said the simple fix was:

```
Installing a new set of root certs does the trick, FYI:

conda install -c conda-forge certifi
```

Can you check if `pip install -U certifi` will fix it?

**Comment by kimishpatel** [comment]
@soumith here? docker/caffe2/jenkins/common/install_python.sh.

**Comment by jamesr66a** [comment]
@soumith I went down this line of reasoning the first time I fixed it (https://github.com/pytorch/pytorch/pull/26062)

tl;dr We should just switch to requests across the board, since we will be consistent in python2/python3, and we will not have to worry about cert validation at all, since `requests` takes a `certifi` dependency and properly uses that CA file

**Review by jamesr66a — APPROVED** [review]

**Review by soumith — CHANGES_REQUESTED** [review]
if you are doing this, you have to change this line as well: https://github.com/pytorch/pytorch/blob/master/setup.py#L356

Or else nightlies tomorrow will fail.

**Comment by kimishpatel** [comment]
Thanks for pointing out @soumith

**Review by soumith — APPROVED** [review]

**Comment by kimishpatel** [comment]
Going to abandon this, as the issue is fixed by conda upstream.

**Comment by ailzhang** [comment]
FYI the conda issue has been resolved so this PR is no longer applicable. Closing...
